From dd0d6ec2de81df09f4f198ede8d6c141cb7b4b0b Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Sun, 23 Aug 2026 12:23:48 +0300
Subject: [PATCH 1/1] build_templates: marvell blacklist with kempld

Expand into build_templates basic rootfs with blacklist file
/etc/modprobe.d/50-kempld.conf specifically for
Marvell AMD64 Kontron/kemld modules wich lead to
- occasional i2c SMBUS and KEMPLD bus enumeration swap with further PMON failure
- I2c-mux collision and bus-stuck on a Marvell device

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 files/build_templates/sonic_debian_extension.j2 | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/files/build_templates/sonic_debian_extension.j2 b/files/build_templates/sonic_debian_extension.j2
index 706a46da3..63f71b0eb 100644
--- a/files/build_templates/sonic_debian_extension.j2
+++ b/files/build_templates/sonic_debian_extension.j2
@@ -924,6 +924,19 @@ fi
 ## copy blacklist file
 sudo cp $IMAGE_CONFIGS/platform/linux_kernel_bde.conf $FILESYSTEM_ROOT/etc/modprobe.d/
 
+# Marvell AMD64 blacklist with Kontron/kemld modules wich lead to
+# - occasional i2c SMBUS and KEMPLD bus enumeration swap with further PMON failure
+# - I2c-mux collision and bus-stuck on a Marvell device
+if [[ "$CONFIGURED_PLATFORM" == "marvell-prestera" && "$CONFIGURED_ARCH" == "amd64" ]]; then
+FILE="$FILESYSTEM_ROOT/etc/modprobe.d/50-kempld.conf"
+sudo mkdir -p "$(dirname "$FILE")"
+grep -q kempld_core "$FILE" 2>/dev/null || sudo tee -a "$FILE" > /dev/null <<'EOF'
+blacklist kempld_core
+blacklist gpio_kempld
+blacklist i2c_kempld
+EOF
+fi
+
 # Enable psample drivers to support sFlow 
 # vpp samples in its own dataplane and needs only the psample netlink channel
 # vs additional needs act_sample for the kernel tc(1) sampler.
-- 
2.34.1

